Sources/ProtobufferCoding/Helpers.swift

```
@@ -8,20 +8,36 @@
 
 import Foundation
 
-/// A helper type conforming to `CodingKey` that does not support string-based coding keys,
+/// A helper type conforming to `CodingKey` that does not necessarily support string-based coding keys,
 /// and will always return the same `intValue`.
 struct FixedCodingKey: CodingKey {
-    /// Guaranteed to be non-nil, but has to be nullable to satisfy the `CodingKey` protocol
-    let intValue: Int?
+    private let _intValue: Int
+    private let _stringValue: String?
     
     init(intValue: Int) {
-        self.intValue = intValue
+        self._intValue = intValue
+        self._stringValue = nil
+    }
+    
+    init(intValue: Int, stringValue: String) {
+        self._intValue = intValue
+        self._stringValue = stringValue
     }
     
     init?(stringValue: String) {
         fatalError("Not supported. Provide an integer")
```

Instead of fataling out, can we use the fact, that this init is declared as optional?
Meaning, just setting `self = nil` and rely on the party using it to generate the error message (which potentially includes more context?). I think, doing so could avoid similar errors as the once we are fixing right now (were some StringConvertible conformance access the properties).
For reference: https://developer.apple.com/documentation/swift/codingkey/2893595-init.

---

We could, but since this is an internal type and only intended to be used with a non-nil int value, i figured it'd be the best approach to make sure you can in fact only use it that way.
It would of course be possible to delegate error handling here to the caller, but i don't see how there is any more context that could be added. Either an int value is provided or not. (IMO the error we're fixing right now is in fact an argument in favour of using a crashing `fatalError` instead of just returning `nil`.)